aapcs64/aapcs64.rst

```diff
@@ -1729,6 +1738,69 @@ it induces ``memset`` to modify ``BLK`` before ``memset`` has returned:
    TPIDR2_EL0 = &BLK;
    memset(&BLK, 0, 16);          // Non-conforming
 
+Asynchronous transfers of control
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**(Beta)**
+
+The specification of the lazy save scheme in this document says that
+certain things must be true of a thread at all times after initialization,
+rather than requiring those things to be true only at function-call
+boundaries.  As a result, the specification also restricts the order
+in which certain state transitions can happen.
+
+This is done to support asynchronous transfers of control, as provided
+by things like POSIX signals.  The exact treatment of such transfers
+is platform-specific and is outside the scope of this specification.
+However, in the case where the asynchronous transfer involves calling
+some form of handler, a platform must implement transfers to and from
+the handler in a way that is compatible with the lazy save scheme.
+This applies even if the handler returns abnormally, such as by calling
+``longjmp`` or by raising an exception, if the platform allows such
+abnormal returns.
+
+One approach that a platform can take for such handlers is as follows:
+
+* Save all SME state before entering a handler.
+
+* Leave a record on the stack so that an unwinder can recover this
+  saved information.
+
+* After saving the SME state, set PSTATE.ZA and TPIDR2_EL0 to zero,
+  so that the handler starts with ZA in the “off” state.
+
+* On a normal return from the handler, restore the saved SME state
+  and resume execution.
+
+* Make ``setjmp`` and ``longjmp`` turn ZA off; see `setjmp and longjmp`_
+  for details.
+
+* When unwinding out of a handler, use the saved TPIDR2_EL0
+  information to see whether ZA was previously dormant; that is,
+  if TPIDR2_EL0 pointed to a TPIDR2 block that has a nonnull ZA save
+  buffer.  If so, copy the saved ZA contents to this buffer.
```

Comment:
Should this also mention that the unwinder must commit the lazy-save and turn ZA off?

Reply:
In that case, would be useful to amend it with something like `, followed by any operations to implement the existing requirements for unwinding exceptions, as described in 'Exceptions'`, or to slightly rephrase the sentence below ("See 'Exceptions' for more information) to make it clear that the above steps are additive to the existing requirements?

The reason I ask, is because that wasn't immediately clear to me from reading the text.